code_neural_networks/scripts/encoderEvaluation.py

```diff
@@ -0,0 +1,523 @@
+        list_hash = []
+        for hash in list_hash_validation:
+            if hash in annotations.keys():
+                list_hash.append(hash)
+
+        num_model = len(list_hash)
+        num_images_per_model = len(annotations[list_hash[0]])
+        num_model_2_render= min(num_images_per_model, args.num_image)
+
+        resolution = args.resolution
+
+        # compute latent codes
+        print("load grid...")
+        grid = load_grid(list_hash, annotations, num_model_2_render, param["image"], param["network"])
+        print("compute code from grid...")
+        latent_code = get_code_from_grid(grid, param_all["latent_size"])
+
+        
+        # fill a xyz grid to give as input to the decoder 
+        xyz = init_xyz(resolution)
+
+        # load decoder
+        decoder = torch.load(DECODER_PATH).cuda()
+        decoder.eval()
+
+        # list to keep in memory all the loss computed
+        list_sdf = []
+        list_rgb = []
+        list_lab = []
+        list_l2 = []
+        list_cham_sdf = []
+        list_cham_rgb = []
+        list_cham_lab = []
+
+        for model_hash, model_id in zip(list_hash, range(num_model)):
+
+            print("\n")
+
+            # decode
+            sdf_target = np.empty([resolution, resolution, resolution, 4])
+
+            for x in range(resolution):
+
+                sdf_pred = decoder(dict_hash_2_code[model_hash].repeat(resolution * resolution, 1).cuda(),xyz[x * resolution * resolution: (x+1) * resolution * resolution]).detach()
+
+                sdf_pred[:,0] = sdf_pred[:,0] * resolution
+                sdf_pred[:,1:] = torch.clamp(sdf_pred[:,1:], 0, 1)
+                sdf_pred[:,1:] = sdf_pred[:,1:] * 255
+
+                sdf_target[x, :, :, :] = np.reshape(sdf_pred[:,:].cpu(), [resolution, resolution, 4])
+
+            print('Minimum and maximum value: %f and %f. ' % (np.min(sdf_target[:,:,:,0]), np.max(sdf_target[:,:,:,0])))
+            if(np.min(sdf_target[:,:,:,0]) < 0 and np.max(sdf_target[:,:,:,0]) > 0):
+                vertices_target, faces_target = marching_cubes(sdf_target[:,:,:,0])
+                colors_v_target = exctract_colors_v(vertices_target, sdf_target)
+                colors_f_target = exctract_colors_f(colors_v_target, faces_target)
+                off_file = "%s/%s_target.off" %(OUTPUT_DIR, model_hash)
+                write_off(off_file, vertices_target, faces_target, colors_f_target)
+                print("Wrote %s_target.off" % model_hash)
+            else:
+                print("surface level: 0, should be comprise in between the minimum and maximum value")
+
+            vertices_target = torch.tensor(vertices_target.copy())
+            colors_v_target = torch.tensor(colors_v_target/255).unsqueeze(0).cuda()
+
+            ########################################################### CODE FOR RENDERING ALL IMAGES ############################################################
+
+            # for j in range(num_model_2_render):
```

Comment:
why is that code commented out?
if it is not useful, delete it.
If it might be useful, uncomment it and place it behind a if (). Moving it into its own function would make sense I guess.